src/iceberg_extension.cpp

```diff
@@ -98,12 +98,21 @@ static unique_ptr<Catalog> IcebergCatalogAttach(StorageExtensionInfo *storage_in
 			endpoint = StringUtil::Lower(entry.second.ToString());
 			StringUtil::RTrim(endpoint, "/");
 		} else {
-			throw BinderException("Unrecognized option for PC attach: %s", entry.first);
+			throw BinderException("Unrecognized option for iceberg catalog attach: %s", entry.first);
 		}
 	}
 	auto warehouse = info.path;
 	auto catalog_type = ICEBERG_CATALOG_TYPE::INVALID;
 
+	// Lookup a secret we can use to access the rest catalog.
+	// if no secret is referenced, this throw
+	auto secret_entry = IRCatalog::GetSecret(context, secret_name);
```

Comment:
We are working on refactoring some of our iceberg secret functionality, we don't want to throw if no secret is available, but rather if no iceberg secret is available. This will also require some changes to the `GetSecret` function, I'll come back to this

Reply:
This is an interesting observation though

If glue/s3tables, we look up the s3 secret
But then in the non-glue/s3tables path, we *also* get the s3 secret

So this is essentially just a deduplication of the current logic.
But we shouldn't be looking up an s3 secret in the non-glue/s3tables path, as Tom said, we should be looking for an `iceberg` secret instead.
I have already started working on a branch for this.

The s3 secret in the non-glue/s3tables path is currently used to look up a key+secret to provide the `/v1/oauth/tokens` endpoint - in which case the key+secret is not actually an s3 secret.
Which is why we're changing this.